packages/loopring_v3/contracts/impl/operators/BatchOperator.sol

```diff
@@ -0,0 +1,137 @@
+/*
+
+  Copyright 2017 Loopring Project Ltd (Loopring Foundation).
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+pragma solidity ^0.5.11;
+pragma experimental ABIEncoderV2;
+
+import "../../iface/IExchangeV3.sol";
+
+import "../../lib/AddressUtil.sol";
+import "../../lib/Claimable.sol";
+import "../../lib/ERC20.sol";
+import "../../lib/ERC20SafeTransfer.sol";
+
+
+/// @title  BatchOperator
+/// @dev    This contract can act as a DEX operator to batch the following ExchangeV30
+///         transactions to save some gas (~10% according to Brecht Devos):
+///            - commitBlock
+///            - verifyBlocks
+///            - revertBlock
+///            - distributeWithdrawals
+///
+/// @author Daniel Wang - <daniel@loopring.org>
+contract BatchOperator is Claimable
+{
+    using ERC20SafeTransfer for address;
+    using AddressUtil       for address;
+
+    event Drained(address indexed token, uint amount);
+
+    address public exchange;
+
+    constructor(IExchangeV3 _exchange)
+        Claimable()
+        public
+    {
+        setExchange(_exchange);
+    }
+
+    function setExchange(IExchangeV3 _exchange)
+        public
+        onlyOwner
+    {
+        exchange = address(_exchange);
+    }
+
+    function batchCall(
+        bytes[] calldata transactions
+        )
+        external
+        payable
+        onlyOwner
+    {
+        require(transactions.length > 0, "EMPTY_DATA");
+        require(msg.value == 0, "INVALID_MSG_VALUE");
+
+        for (uint i = 0; i < transactions.length; i++) {
+            /* solium-disable-next-line */
+            (bool success, ) = exchange.call(transactions[i]);
+            require(success, "FAILURE");
+        }
+    }
+
+    function batchCall(
```

Comment:
Seems like it would be better to just to have this `batchcall` so we don't have 2 similar functions (with one that can just do less)?

Reply:
Removed